accepted/future-releases/constructor-tearoffs/feature-specification.md

```diff
@@ -354,21 +403,43 @@ f(x.a<b,c>-d);  // f((x.a<b, c>)-d) or f((x.a < b), (c > -d]))
+We will disambiguate such situations *heuristically* based on the token following the `>`. In the existing ambiguity we treat `(` as a sign that it's a generic invocation. We extend the number of tokens which, when following a potential type argument list, makes us choose to parse the previous tokens as that type argument list. 
 
-The look-ahead tokens which force the prior tokens to be type arguments are:
+There is a number of tokens which very consistently *end* an expression, and we include all those:
 
-> `(`  `)`  `]`  `}`  `:`  `;`  `,`  `.`  `?`  `==`  `!=` `..` `?.` `??` `?..` 
->
-> `&` `|` `^` `+` `*`  `%`  `/`  `~/` `is` `as`
+> `)`, `}`, `]`, `;`, `:`, `,`
 
-Any other token following the ambiguous `>` will make the prior tokens be parsed as comma separated `<` and `>` operator invocations.
+The we include tokens which we *predict* will continue a generic instantiation:
 
-_We could add `&&` and `||` to the list, but it won't matter since the result is going to be invalid in either case, because those operators do not work on `Type` or function values and cannot be defined using extension methods._
+>  `(`  `.`  `==`  `!=` 
 
-_This might set us up for problems if we ever decide to use any of the infix operators as prefix operators, like `-` is now, but it does allow defining those operators on `Type` or `Function` and using them. Not allowing the infix operators is an alternative_
+The first six are tokens which cannot possibly start an expression, and therefore cannot occur after a greater-than infix operator. The last four tokens can continue an expression, and of those only `(` can also start an expression, and we already decided how to disambiguate that).
 
-_Note that including `as` in this list is technically a breaking change because `as` is not a reserved word, but this is extremely unlikely to cause a breakage in practice._
+There are many other tokens which *currently* cannot continue an expression (and therefore cannot validly follow a type argument list) or which cannot *start* an expression (and therefore cannot validly follow a greater-than operator), but in the service of keeping our future options open, we do not choose to depend those at the current time. For example we omit most infix operators, even though they *currently* cannot start a new expression, and therefore cannot follow a `>` infix operator. This leaves us open to allowing some of those operators as prefix operators in the future, like we currently allow `-`.
```

Comment:
I'm not sure what "we do not choose to depend those at the current time" means.  Did you mean "we do not chose to include those at the current time"?

Also, at the end of the paragraph, "... like we currently alllow `-`." feels incomplete.  IMHO it would be clearer to say "... like we currently allow with `-`."

Reply:
I probably meant "to depend on" as in "depending on those not being possible expression starters". I'll expand it.

Reply:
I can live with the updated text, but IMHO it represents the situation a little inaccurately, because it seems to imply that if we included, say `+`, as a continuation token, we would necessarily be precluded from introducing new syntax in a future release of the language that allowed `+` to follow a `>` operator.  The situation is not really that bad.  The issue is simply that if we included `+` as a continuation token, then code like `f(a < b, c > + d)` would be accepted by the parser today as equivalent to `f((a<b, c>) + d)`, and then in the future if we introduced prefix `+`, the parser would continue to treat `f(a < b, c > + d)` as `f((a<b, c>) + d)`, rather than `f((a < b), (c > +d))`, possibly confusing people in some rare circumstances.

I'm not arguing with the decision, just saying that the rationale for it is not as dire as this paragraph makes it seem.